utils/zerovec/src/ule/mod.rs

```diff
@@ -285,9 +313,13 @@ pub unsafe trait VarULE: 'static {
     ///
     /// The default implementation executes `Self::validate_byte_slice` followed by
     /// `Self::from_byte_slice_unchecked`.
-    fn parse_byte_slice(bytes: &[u8]) -> Result<&Self, Self::Error> {
-        Self::validate_byte_slice(bytes)?;
-        Ok(unsafe { Self::from_byte_slice_unchecked(bytes) })
+    fn parse_byte_slice(bytes: &[u8]) -> Result<&Self, UleError<Self::Error>> {
+        Self::validate_byte_slice(bytes).map_err(|e| UleError::InvalidBytes(e))?;
+        let result = unsafe { Self::from_byte_slice_unchecked(bytes) };
+        if mem::size_of_val(result) != mem::size_of_val(bytes) {
```

Comment:
issue: similarly I feel like this should be a `debug_assert`, and it's `validate_byte_slice`'s job to validate this. This way we don't need a separate error type

Reply:
Done.